starky/src/traits.rs

```diff
@@ -35,8 +42,59 @@ pub trait MerkleTree {
 
 pub trait Transcript {
     fn new() -> Self;
-    fn get_field(&mut self) -> F3G;
+    fn get_field<F: FieldExtension>(&mut self) -> F;
     fn get_fields1(&mut self) -> Result<FGL>;
     fn put(&mut self, es: &[Vec<FGL>]) -> Result<()>;
     fn get_permutations(&mut self, n: usize, nbits: usize) -> Result<Vec<usize>>;
 }
+
+pub trait FieldExtension:
+    From<Fr>
+    + From<u64>
+    + From<i32>
+    + From<usize>
+    + Debug
+    + Hash
+    + Copy
+    + Clone
+    + PartialEq
+    + Eq
+    + Default
+    + Add<Output = Self>
+    + AddAssign
+    + Div<Output = Self>
+    + DivAssign
+    + Mul<Output = Self>
+    + MulAssign
+    + Neg<Output = Self>
+    + Sub<Output = Self>
+    + SubAssign
+    + Rand
+    + Display
+    + Send
+    + Sync
+    + Field
+    + Serialize
+{
+    const ELEMENT_BYTES: usize;
+    const IS_CANONICAL: bool = false;
+    const ZERO: Self;
+    const ONE: Self;
+    const NEW_SIZE: u64 = 0;
+    fn dim(&self) -> usize;
+    fn from_vec(values: Vec<Fr>) -> Self;
+    fn to_be(&self) -> Fr;
+    fn as_elements(&self) -> Vec<Fr>;
+    fn mul_scalar(self, b: usize) -> Self;
+    fn eq(self, rhs: &Self) -> bool;
+    fn gt(self, rhs: &Self) -> bool;
+    fn geq(self, rhs: &Self) -> bool;
+    fn lt(self, rhs: &Self) -> bool;
+    fn leq(self, rhs: &Self) -> bool;
+    fn exp(self, e_: usize) -> Self;
+    fn batch_inverse(elems: &[Self]) -> Vec<Self>;
```

Comment:
Actual `batch_inverse` is not a property of a Field.
This should be implment outsize the trait.

Reply:
As comments above, those comparators are not really used, so the first argument is declared as self, which should be &self.

Reply:
already moved the batch_inverse outside of the FieldExtension